From 1b9211793eba7bc6a0b0857d357d4d8cfca2c387 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Thu, 24 Feb 2011 03:14:35 +0100
Subject: [PATCH] Revert "Bluetooth: Enable USB autosuspend by default on btusb"

This reverts commit 556ea928f78a390fe16ae584e6433dff304d3014.

On Wed, Feb 23, 2011 at 04:25:23PM -0800, Linus Torvalds wrote:
> Hmm. Is there any reason we shouldn't revert commit
> 556ea928f78a390fe16ae584e6433dff304d3014 given the regression?

I think reverting makes sense at the moment. I haven't seen the reported
issues - I need to figure out if this is related to bus powered/self
powered devices, or what else is causing the difference.

> It apparently had problems before too, and caused autosuspend to be
> disabled entirely, judging at least by
>
>   https://bugzilla.redhat.com/show_bug.cgi?id=528744
>
> but there's obviously the comment about "those should be fixed now".
> Apparently there are more issues.

Right, we were autosuspending even when there were active connections.
Marcel told me that ought to be fixed and my testing seemed to agree...

> I have no idea whether this is a USB-level issue, or a driver-level
> one. There are no comments about exactly what fixed the input device
> issues. So I'm adding both BT and USB people to the discussion.

I'll try to get time to look at this this week. Marcel, any ideas?

Matthew Garrett | mjg59@srcf.ucam.org
---
 drivers/bluetooth/btusb.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -1038,8 +1038,6 @@ static int btusb_probe(struct usb_interf
 
 	usb_set_intfdata(intf, data);
 
-	usb_enable_autosuspend(interface_to_usbdev(intf));
-
 	return 0;
 }
 
